qbft/controller.go

```diff
@@ -82,10 +79,12 @@ func (c *Controller) ProcessMsg(msg *SignedMessage) (*SignedMessage, error) {
 	if IsDecidedMsg(c.Share, msg) {
 		return c.UponDecided(msg)
 	} else if c.isFutureMessage(msg) {
```

Comment:
why not remove the condition?

Reply:
The answer is given within the comment in the code:
```
 else if c.isFutureMessage(msg) {
		// No error is returned if msg is future because it is a valid case.
		// The implementation may choose to create optimizations for this case.
		return nil, nil
	}
```

If we delete this segment then the code will move to `c.UponExistingInstanceMsg(msg)` which will not be always true in case a future message does show up...
It can happen if the eth2 node is lagging.


There are 2 approaches we can take:
1. Implicit - simply don't write cases the spec won't handle and let the implementor figure out that the edge case is missing.
2. Explicit - remind the implementor that the spec doesn't cover some edge case.

I think Explicit is better.

Reply:
I'm not totally convinced. 

We shouldn't create "optimization" hocks in spec for implementation. 

Why not return error saying it's a future message, controller can't process?

Reply:
fixed